### PR TITLE
Calculate DirectoryDepth for Directory SAS; uint? to int? DirectoryDepth

### PR DESCRIPTION
**Pull request by amnguye**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Files.DataLake/tests/DataLakeSasBuilderTests.cs [hunk 1]
```diff
@@ -212,15 +212,13 @@ public async Task DataLakeSasBuilder_DirectoryRawPermissions_Exists()
...
-            // Set the Resource = "d" and the DirectoryDepth = 1
             DataLakeSasBuilder dataLakeSasBuilder = new DataLakeSasBuilder
             {
                 StartsOn = Recording.UtcNow.AddHours(-1),
                 ExpiresOn = Recording.UtcNow.AddHours(1),
                 FileSystemName = test.FileSystem.Name,
                 Path = directory.Path,
-                IsDirectory = true,
-                DirectoryDepth = 1
+                IsDirectory = true
```
Comment (gapra-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/tests/DataLakeSasBuilderTests.cs:221]: Did you want to add a test to test that ensure state method with a number of different object paths?

Comment (amnguye) [reply sdk/storage/Azure.Storage.Files.DataLake/tests/DataLakeSasBuilderTests.cs:221]: Yes I did! I guess I erased it before I pushed the PR. Thanks for the catch

**Review by gapra-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.DataLake/tests/DataLakeTestBase.cs [hunk 2]
```diff
@@ -44,7 +44,7 @@ public abstract class DataLakeTestBase : StorageTestBase
                 "user:ec3595d6-2c17-4696-8caa-7e139758d24a,group:ec3595d6-2c17-4696-8caa-7e139758d24a," +
                 "default:user:ec3595d6-2c17-4696-8caa-7e139758d24a,default:group:ec3595d6-2c17-4696-8caa-7e139758d24a");
         public DataLakeTestBase(bool async, DataLakeClientOptions.ServiceVersion serviceVersion, RecordedTestMode? mode = null)
-            : base(async, mode)
+            : base(async, RecordedTestMode.Record)
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/tests/DataLakeTestBase.cs:47]: Switch back to `mode`.  I've been doing this a lot recently, too.

File: sdk/storage/Azure.Storage.Files.DataLake/src/Sas/DataLakeSasBuilder.cs [hunk 3]
```diff
@@ -481,6 +481,16 @@ internal void EnsureState()
                 else
                 {
                     Resource = Constants.Sas.Resource.Directory;
+                    if (!Path.Equals("/", StringComparison.InvariantCultureIgnoreCase))
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/src/Sas/DataLakeSasBuilder.cs:484]: I think maybe we should add a unit test with a couple `TestCase()`s for directory depth calculation.

**Review by seanmcc-msft — APPROVED** [review]

**Review by seanmcc-msft — APPROVED** [review]

**Pull request merged**